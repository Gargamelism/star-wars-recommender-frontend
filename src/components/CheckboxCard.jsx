import React from "react";

export default function CheckboxCard({ isChecked, name, value, handleChecked }) {
    return (
        <div key={name} className="flex justify-around items-center border-2 rounded border-gray-400 m-2 w-full md:w-96">
            <input type="checkbox" value={value} checked={isChecked} onChange={handleChecked} className="flex-none w-8" />
            <span className="flex-grow text-center">{name}</span>
        </div>
    );
}