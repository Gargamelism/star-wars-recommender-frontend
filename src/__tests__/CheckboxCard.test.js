import renderer from "react-test-renderer";
import CheckboxCard from "../components/CheckboxCard";

test("renders CheckboxCard", () => {
  const tree = renderer
    .create(<CheckboxCard isChecked={false} name="1222" value="222" handleChecked={(val)=>(val)} />)
    .toJSON();
    
    expect(tree).toMatchSnapshot();  
});
