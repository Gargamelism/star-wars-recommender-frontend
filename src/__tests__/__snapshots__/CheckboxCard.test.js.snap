// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CheckboxCard 1`] = `
<div
  className="flex justify-around items-center border-2 rounded border-gray-400 m-2 w-full md:w-96"
>
  <input
    checked={false}
    className="flex-none w-8"
    onChange={[Function]}
    type="checkbox"
    value="222"
  />
  <span
    className="flex-grow text-center"
  >
    1222
  </span>
</div>
`;
